documentation/bolt_types_reference.md

```diff
@@ -0,0 +1,150 @@
+# Bolt data types
+
+This page lists the Bolt data types and describes how to use them.
+
+## `ApplyResult`
+
+An [apply action](applying_manifest_blocks.md#return-value-of-apply-action)
+returns an `ApplyResult`. An `ApplyResult` is part of a `ResultSet` object and
+contains information about the apply action. You can access `ApplyResult`
+functions with dot notation, using the syntax: `$apply_result.function`.
+
+For example, the following plan uses an apply action to install Docker as root,
```

Comment:
At least to me this seems like it breaks the docs vs. reference distinction. I don't know if that's something formalized, but imo we should have examples and explanations in the docs proper, and this doc should assume that users already know how to use `apply` for example. Open to discussing it though.